adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/AmqpContext.java

```diff
@@ -194,4 +201,26 @@ private ErrorCondition getErrorCondition(final ServiceInvocationException error)
             return ProtonHelper.condition(AmqpError.PRECONDITION_FAILED, error.getMessage());
         }
     }
+
+    /**
+     * Validates the address contained in an AMQP 1.0 message property.
+     * 
+     * @param to The property to validate.
+     * @return A future indicating the outcome of the validation operation.
+     */
+    Future<Void> validateMessageProperty(final String to) {
```

Comment:
this method should not return a Future. The instance created by the constructor should be initialized once the constructor returns, shouldn't it?
why not simply return void and declare a `ServiceInvocationException` being thrown in case of a failing check?